intro-to-functions/practice-intro-to-functions.md

```diff
@@ -0,0 +1,935 @@
+<!-- prettier-ignore-end -->
+
+### In Your Own Words
+
+<!-- Question 12 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: short-answer
+* id: 0b46033a-b21c-442a-9238-5c03ca365bb0
+* title: Intro to Functions
+
+##### !question
+
+Using your own words, define what "passing something into a function" means.
+
+##### !end-question
+##### !placeholder
+
+Passing a function means...
+
+##### !end-placeholder
+##### !answer
+
+/.*/
+
+##### !end-answer
+##### !explanation
+
+An example description: Passing something into a function means that we are calling a function and supplying an argument to it. We pass in values as arguments. We can also say that the values we expect to be passed into a function are the parameters. We can say that "passing something into a function" is giving "input" into the function.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 13 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: short-answer
+* id: BrEpeA
+* title: Intro to Functions
+
+##### !question
+
+Using your own words, define what "getting a return value from a function" means.
+
+##### !end-question
+##### !placeholder
+
+Getting a return value from a function means...
+
+##### !end-placeholder
+##### !answer
+
+/.*/
+
+##### !end-answer
+##### !explanation
+
+An example description: Getting a return value from a function means receiving whatever that function returns. After we get a return value from a function call, we usually assign it to a variable. When we invoke a function, and it evaluates into one value, using that one value is "getting its return value." We can say that "getting a return value" is getting "output" from the function.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Syntax Reordering
+
+<!-- Question 14 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: ordering
+* id: nFr02Z
+* title: Intro to Functions
+##### !question
+
+Of the lines of code below, re-order them to its ideal order. Assume that each line would be indented correctly.
+
+##### !end-question
+##### !answer
+
+1. `def sing_im_a_little_teapot():`
+1. `line_1 = "I'm a little teapot"`
+1. `line_2 = "Short and stout"`
+1. `line_3 = "Here is my handle"`
+1. `line_4 = "Here is my spout"`
+1. `entire_song = f"{line_1}, {line_2}. {line_3}, {line_4}!"`
+1. `return entire_song`
+
+##### !end-answer
+##### !explanation
+
+This is the ideal construction of this function:
+
+```python
+def sing_im_a_little_teapot():
+    line_1 = "I'm a little teapot"
+    line_2 = "Short and stout"
+    line_3 = "Here is my handle"
+    line_4 = "Here is my spout"
+    entire_song = f"{line_1}, {line_2}. {line_3}, {line_4}!"
+    return entire_song
+```
+
+Technically, we could re-orer the assignments of `line_1`, `line_2`, `line_3`, and `line_4` to any combination we want. However, it's ideal to have them ordered because of readability and coherence to all programmers. :)
```

Comment:
```suggestion
Technically, we could order the assignments of `line_1`, `line_2`, `line_3`, and `line_4` any way we want. However, we should prefer putting them into an order that minimizes potential confusion for other programmers who read it.
```

Just taking your first paragraph; not because I disagree with the second paragraph or don't think it's worth teaching, but for the sake of making practice worksheets a slim slippery smooth worksheet that only teaches the bare minimum of new content and coding style (such as the Unicode thing above. We could've definitely added more content, but we only added enough so they could get through the question)

I'd like to separate worksheets from teaching code style. In practice worksheets, right now my current teaching belief is that students should freely use these to write really messy code if they'd like, and that's what they got in order to get through it. They'll get better and pick up code style stuff like this from talking to others, reading more code (including the sample answers we give them), sharing code, etc etc etc

Reply:
Sure I could also say that the second paragraph is less of a code style thing but /shrug. Just not the energy I want to pull into these worksheets.

Reply:
It's true that minimizing the surprise for other developers is the most salient and useful take away. Arguably, the second paragraph detracts from that.